src/useFetch.js

```diff
@@ -0,0 +1,44 @@
+import { useState, useEffect } from "react";
+
+export const useFetch = (url) => {
+  const [data, setData] = useState(null);
+  const [isPending, setIsPending] = useState(false);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    const controller = new AbortController();
```

Comment:
is this for the loading spinner?

Reply:
That whole file is the custom hook that Alex shared, I made a small change to it to purposely delay the fetching of data and make it easier to see the loading spinner, but the hook itself is not related to the spinner.